service/controller/release/resource/apps/resource.go

```diff
@@ -105,6 +122,18 @@ func (r *Resource) ensureState(ctx context.Context) error {
 
 	appsToCreate := calculateMissingApps(components, apps)
 	for _, app := range appsToCreate.Items {
+		cmRef, secretRef := key.GetAppConfig(app, configs)
+		if cmRef.Name == "" && secretRef.Name == "" {
```

Comment:
That does not happen. App status and CM and Secret ref are set in a single request in config-controller. But I'm ok to keep that logic.